File: packages/management/src/api/menu.ts
import { secured } from './client';

export type MenuItem = {
  id: number;
  name: string;
  category: string;
  storePrice: number;
  storePriceAvailable: boolean;
  linemanPrice: number;
  linemanPriceAvailable: boolean;
  linemanCostPrice: number;
  costPrice: number;
  status: 'available' | 'soldout';
  recipeStatus?: 'ready' | 'missing_recipe' | 'insufficient_stock';
  sellable?: boolean;
  ingredients: {
    inventoryItemId: number;
    name: string;
    quantity: number;
    unit: string;
    costAmount: number;
  }[];
  imageUrl: string;
  preparationSteps?: string;
};

export type MenuInput = {
  name: string;
  category: string;
  storePrice: number;
  linemanPrice: number;
  linemanCostPrice: number;
  costPrice: number;
  ingredients: { inventoryItemId: number; quantity: number; unit: string }[];
  preparationSteps?: string;
};

export const listMenuItems = (branchCode = 'SBC-AYA-001') =>
  secured<MenuItem[]>(
    `/menu-items?branchCode=${encodeURIComponent(branchCode)}`,
  );

export const createMenuItem = (data: MenuInput, branchCode: string) =>
  secured<{ id: number }>(
    `/menu-items?branchCode=${encodeURIComponent(branchCode)}`,
    { method: 'POST', data },
  );

export const updateMenuItem = (
  id: number,
  data: MenuInput,
  branchCode: string,
) =>
  secured<{ id: number }>(
    `/menu-items/${id}?branchCode=${encodeURIComponent(branchCode)}`,
    { method: 'PATCH', data },
  );

File: apps/api/internal/model/menu.go
package model

import "time"

type MenuStatus string

const (
	MenuStatusAvailable MenuStatus = "available"
	MenuStatusSoldOut   MenuStatus = "soldout"
)

type MenuItem struct {
	ID                    int64            `json:"id"`
	BranchID              int64            `json:"branchId"`
	Name                  string           `json:"name"`
	Category              string           `json:"category"`
	StorePrice            float64          `json:"storePrice"`
	StorePriceAvailable   bool             `json:"storePriceAvailable"`
	LinemanPrice          float64          `json:"linemanPrice"`
	LinemanPriceAvailable bool             `json:"linemanPriceAvailable"`
	LinemanCostPrice      float64          `json:"linemanCostPrice"`
	CostPrice             float64          `json:"costPrice"`
	Status                MenuStatus       `json:"status"`
	ImageURL              string           `json:"imageUrl"`
	Ingredients           []MenuIngredient `json:"ingredients,omitempty"`
	PreparationSteps      string           `json:"preparationSteps"`
	CreatedAt             time.Time        `json:"createdAt,omitempty"`
	UpdatedAt             time.Time        `json:"updatedAt,omitempty"`
}
